src/pyspiffe/spiffe_id/trust_domain.py

```diff
@@ -56,9 +56,9 @@ def __set_name(self, name: str) -> None:
             )
 
         name = self.normalize(name)
-        uri = urlparse(name)
+        uri: ParseResult = urlparse(name)
         self.validate_uri(uri)
-        self.__name = uri.hostname.lower().strip()
+        self.__name = cast(str, uri.hostname).lower().strip()
```

Comment:
The error here was: `error: Item "None" of "Optional[str]" has no attribute "lower"`
`self.validate_uri(uri)` already checks if `hostname` is None and raise an exception in that case. Mypy is no able to detect that so it complains about `hostname` been `None`.
A way to solve that is using `Typing.cast(type, value)`.
In other places it could make sense to directly have something like:
```
        uri: ParseResult = urlparse(name)
        if not uri.hostname:
            raise ValueError(' ')
        self.__name = uri.hostname.lower().strip()
```
And in that case Mypy won't complain.

Reply:
Thanks for the explanation, I was a bit confused why mypy was complaining about this.